pkg/cluster/bootstrap.go

```diff
@@ -29,29 +34,156 @@ func (c *Cluster) Bootstrap(ctx context.Context) error {
+		if isInitialized && !isHTTP {
+			tmpDataDir := filepath.Join(c.config.DataDir, "db", "tmp-etcd")
+			os.RemoveAll(tmpDataDir)
+			if err := os.Mkdir(tmpDataDir, 0700); err != nil {
+				return err
+			}
+			etcdDataDir := etcd.DBDir(c.config)
+			if err := createTmpDataDir(etcdDataDir, tmpDataDir); err != nil {
+				return err
+			}
+
+			args := executor.ETCDConfig{
+				DataDir:           tmpDataDir,
+				ForceNewCluster:   true,
+				ListenClientURLs:  "http://127.0.0.1:2399",
+				Logger:            "zap",
+				HeartbeatInterval: 500,
+				ElectionTimeout:   5000,
+				LogOutputs:        []string{"stderr"},
+			}
+			configFile, err := args.ToConfigFile()
+			if err != nil {
+				return err
+			}
+			cfg, err := embed.ConfigFromFile(configFile)
+			if err != nil {
+				return err
+			}
+
+			etcd, err := embed.StartEtcd(cfg)
+			if err != nil {
+				return err
+			}
+			defer func() {
+				etcd.Close()
+				if err := os.RemoveAll(tmpDataDir); err != nil {
+					logrus.Warn("failed to remove etcd temp dir", err)
+				}
+			}()
+
+			data, err := c.retrieveInitializedDBdata(ctx)
+			if err != nil {
+				return err
+			}
+
+			ec := endpoint.ETCDConfig{
+				Endpoints:   []string{"http://127.0.0.1:2399"},
+				LeaderElect: false,
+			}
+
+			if err := c.ReconcileBootstrapData(ctx, bytes.NewReader(data.Bytes()), &c.config.Runtime.ControlRuntimeBootstrap, false, &ec); err != nil {
+				logrus.Fatal(err)
+			}
+		}
+	}
+
 	if c.shouldBootstrap {
 		return c.bootstrap(ctx)
 	}
 
 	return nil
 }
 
+// copyFile copies the contents of the src file
+// to the given destination file.
+func copyFile(src, dst string) error {
+	srcfd, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer srcfd.Close()
+
+	dstfd, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer dstfd.Close()
+
+	if _, err = io.Copy(dstfd, srcfd); err != nil {
+		return err
+	}
+
+	srcinfo, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+
+	return os.Chmod(dst, srcinfo.Mode())
+}
+
+// createTmpDataDir creates a temporary directory and copies the
+// contents of the original etcd data dir to be used
+// by etcd when reading data.
+func createTmpDataDir(src, dst string) error {
+	srcinfo, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+
+	if err := os.MkdirAll(dst, srcinfo.Mode()); err != nil {
+		return err
+	}
+
+	fds, err := ioutil.ReadDir(src)
+	if err != nil {
+		return err
+	}
+
+	for _, fd := range fds {
+		srcfp := path.Join(src, fd.Name())
+		dstfp := path.Join(dst, fd.Name())
+
+		if fd.IsDir() {
+			if err = createTmpDataDir(srcfp, dstfp); err != nil {
+				fmt.Println(err)
+			}
+		} else {
+			if err = copyFile(srcfp, dstfp); err != nil {
+				fmt.Println(err)
+			}
+		}
+	}
+
+	return nil
+}
+
 // shouldBootstrapLoad returns true if we need to load ControlRuntimeBootstrap data again.
 // This is controlled by a stamp file on disk that records successful bootstrap using a hash of the join token.
```

Comment:
can you update the comment here to reflect the initialized vs non-initialized return bool?